account_parent/models/account.py

```diff
@@ -0,0 +1,127 @@
+                sum(debit - credit) balance
+        FROM account_move_line l
+        LEFT JOIN account_move m ON l.move_id = m.id
+        LEFT JOIN account_account a ON l.account_id = a.id
+        LEFT JOIN account_account_type at ON a.user_type_id = at.id
+        WHERE 1 = 1 AND l.account_id in %s
+        """
+        res = ''
+        date_field = 'date'
+        if context.get('aged_balance'):
+            date_field = 'date_maturity'
+        if context.get('date_to'):
+            res += " AND l.{} <= '{}' ".format(date_field, context['date_to'])
+        if context.get('date_from'):
+            if not context.get('strict_range'):
+                res += " AND ({} >= '{}' ".format(
+                    date_field, context['date_from'], )
+                res += " OR at.include_initial_balance = true) "
+            elif context.get('initial_bal'):
+                res += " AND l.{} < '{}' ".format(
+                    date_field, context['date_from'])
+            else:
+                res += " AND l.{} >= '{}' ".format(
+                    date_field, context['date_from'])
+        if context.get('journal_ids'):
+            res += " AND l.journal_id in {} ".format(
+                tuple(context['journal_ids']))
+        state = context.get('state')
+        if state and state.lower() != 'all':
+            res += """ AND m.state = '%s' """ % state
+        if context.get('company_id'):
+            res += " AND l.company_id = {} ".format(context['company_id'])
+        if 'company_ids' in context:
+            res += " AND l.company_id  in {} ".format(
+                tuple(context['company_ids']))
+        query += res and res
+        return query
+
+    @api.multi
+    @api.depends('move_line_ids', 'move_line_ids.amount_currency',
+                 'move_line_ids.debit', 'move_line_ids.credit')
+    def _compute_values(self):
+        for account in self:
+            query = account._move_domain_get()
+            sub_accounts = self.with_context(
+                {'show_parent_account': True}).search([
+                    ('id', 'child_of', [account.id])])
+            self.env.cr.execute(query, (tuple(sub_accounts.ids),))
+            result = self.env.cr.dictfetchall()[0]
+            account.balance = result['balance']
+            account.credit = result['credit']
+            account.debit = result['debit']
+
+    move_line_ids = fields.One2many(
+        'account.move.line', 'account_id', 'Journal Entry Lines')
+    balance = fields.Float(
+        compute='_compute_values', digits=dp.get_precision('Account'))
+    credit = fields.Float(
+        compute='_compute_values', digits=dp.get_precision('Account'))
+    debit = fields.Float(
+        compute='_compute_values', digits=dp.get_precision('Account'))
+    parent_id = fields.Many2one(
+        'account.account', 'Parent Account', ondelete='restrict')
+    child_ids = fields.One2many(
+        'account.account', 'parent_id', 'Child Accounts')
```

Comment:
Hi everyone,

Why don't we just implement the Hierarchy implemented in Odoo/master
that just landed a few days ago at commit https://github.com/odoo/odoo/commit/59676002968294a3cf3dbf7ffaef31e14bf00dd4

This new approach, I think could fit to everyone allowing to provide Hierarchy,
Using the current way can lead to huge overheads to compute the recursive
values of the parent-children accounts, I am aware that we need the Hierarchy
but this new approach leaves the computation of the parent-child values to 
reporting and the parent-children relationship is separated into two, that hierarchy
is only left to parents which bear no moves (account.group) and the account.account
is the ultimate children that bears only moves.

As @fpodoo explained at https://github.com/odoo/odoo/pull/17367#issuecomment-310749779

This is more elegant and less power/resource consuming.

This computed fields will lead to the overhead that once crippled the previous odoo versions.

Best Regards

Reply:
+1